src/web/app/pages-instructor/instructor-session-base-page.component.ts

```diff
@@ -192,17 +195,84 @@ export abstract class InstructorSessionBasePageComponent {
-              'The feedback session has been copied. Please modify settings/questions as necessary.',
-              { courseid: createdSession.courseId, fsname: createdSession.feedbackSessionName });
-        }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorToast(resp.error.message); });
+  copySessionFromRowModel(model: SessionsTableRowModel, result: CopySessionResult): Observable<FeedbackSession>[] {
+    const copySessionRequests: Observable<FeedbackSession>[] = [];
+    this.failedToCopySessions = {};
+    result.copyToCourseList.forEach((copyToCourseId: string) => {
+      copySessionRequests.push(
+          this.copyFeedbackSession(model.feedbackSession, result.newFeedbackSessionName, copyToCourseId)
+              .pipe(catchError((err: any) => {
+                this.failedToCopySessions[copyToCourseId] = err.error.message;
+                return of(err);
+              })),
+      );
+    });
+    return copySessionRequests;
+  }
+
+  /**
+   * Creates list of copy session requests from params
+   * @param result the result of the copy session modal
+   * @param courseId the source courseId
+   * @param feedbackSessionName the source feedback session name
+   * @returns the list of copy session requests
+   */
+  copySessionFromModal(result: CopySessionModalResult, courseId: string, feedbackSessionName: string)
+      : Observable<FeedbackSession>[] {
+    const copySessionRequests: Observable<FeedbackSession>[] = [];
+    this.failedToCopySessions = {};
+    result.copyToCourseList.forEach((copyToCourseId: string) => {
+      copySessionRequests.push(this.feedbackSessionsService.getFeedbackSession({
+        courseId,
+        feedbackSessionName,
+        intent: Intent.FULL_DETAIL,
+      }).pipe(switchMap((feedbackSession: FeedbackSession) =>
+              this.copyFeedbackSession(feedbackSession, result.newFeedbackSessionName, copyToCourseId)
+          .pipe(catchError((err: any) => {
+            this.failedToCopySessions[copyToCourseId] = err.error.message;
+            return of(err);
+          })),
+      )));
+    });
+    return copySessionRequests;
+  }
+
+  /**
+   * Submits a single copy session request.
+   */
+  copySingleSession(copySessionRequest: Observable<FeedbackSession>): void {
+    copySessionRequest.subscribe((createdSession: FeedbackSession) => {
+      if (Object.keys(this.failedToCopySessions).length === 0) {
+        this.navigationService.navigateWithSuccessMessage(this.router,
+            '/web/instructor/sessions/edit',
+            'The feedback session has been copied. Please modify settings/questions as necessary.',
+            { courseid: createdSession.courseId, fsname: createdSession.feedbackSessionName });
+      } else {
+        this.statusMessageService.showErrorToast(this.getCopyErrorMessage());
+      }
+    }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorToast(resp.error.message); });
+  }
+
+  resolveCopyRequest(): void {
```

Comment:
This is not exactly resolving the request; this is merely showing status message.